101-vm-ubuntu-DSVM-GPU-or-CPU/azuredeploy.json

```diff
@@ -0,0 +1,311 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "adminUsername": {
+            "type": "string",
+            "metadata": {
+                "description": "Username for Administrator Account"
+            }
+        },
+        "adminPassword": {
+            "type": "secureString",
+            "metadata": {
+                "description": "Password for for Administrator Account"
+            }
+        },
+        "vmName": {
+            "type": "string",
+            "defaultValue": "[concat('DataScienceVM-', uniqueString(utcNow('F')))]",
+            "metadata": {
+                "description": "The name of you Virtual Machine."
+            }
+        },
+        "location": {
+            "type": "string",  
+            "defaultValue": "[resourceGroup().location]",  
+            "metadata": {  
+                    "description": "Location for all resources."  
+            }  
+        },
+        "Cpu-Gpu": {
+            "type": "string",
+            "defaultValue": "cpu",
+            "allowedValues": [
+                "cpu",
+                "gpu"
+            ],
+            "metadata": {
+                "description": "Choose between CPU or GPU processing"
+            }
+        },
+        "virtualNetworkName": {
+            "type": "string",
+            "defaultValue": "vNet",
+            "metadata": {
+                "description": "Name of the VNET"
+            }
+        },
+        "subnetName": {
+            "type": "string",
+            "defaultValue": "subnet",
+            "metadata": {
+                 "description": "Name of the subnet in the virtual network"
+            }
+        },
+        "networkSecurityGroupName": {
+            "type": "string",
+            "defaultValue": "SecGroupNet",
+            "metadata": {
+                "description": "Name of the Network Security Group"
+            }
+        },
+        "virtualMachineSizeCPU": {
+            "type": "string",
+            "defaultValue": "Standard_DS1_v2",
+            "allowedValues": [
+                "Standard_DS1_v2"
+            ],
+            "metadata":{
+                "description": "CPU VM size."
+            }
+        },
+        "virtualMachineSizeGPU": {
+            "type": "string",
+            "defaultValue": "Standard_NC6_Promo",
+            "allowedValues": ["Standard_NC6_Promo"],
+            "metadata":{
+                "description": "GPU VM size."
+            }
+        }
+    },
+    "variables": {
```

Comment:
add var for vm size, e.g.

"vmSize" {
    "cpu": "Standard_DS1_v2",
    "gpu": "Standard_NC6_Promo"
}

Reply:
Created variable